pkg/sql/catalog/tabledesc/structured.go

```diff
@@ -2608,27 +2605,41 @@ func checkColumnsValidForIndex(tableDesc *Mutable, indexColNames []string) error
 		}
 	}
 	if len(invalidColumns) > 0 {
-		return notIndexableError(invalidColumns, false)
+		return notIndexableError(invalidColumns)
 	}
 	return nil
 }
 
 func checkColumnsValidForInvertedIndex(tableDesc *Mutable, indexColNames []string) error {
-	invalidColumns := make([]descpb.ColumnDescriptor, 0, len(indexColNames))
+	lastCol := len(indexColNames) - 1
 	for i, indexCol := range indexColNames {
 		for _, col := range tableDesc.AllNonDropColumns() {
 			if col.Name == indexCol {
-				lastCol := len(indexColNames) - 1
-				if i == lastCol && !colinfo.ColumnTypeIsInvertedIndexable(col.Type) ||
-					i < lastCol && !colinfo.ColumnTypeIsIndexable(col.Type) {
-					invalidColumns = append(invalidColumns, col)
+				// The last column indexed by an inverted index must be
+				// inverted indexable.
+				if i == lastCol && !colinfo.ColumnTypeIsInvertedIndexable(col.Type) {
+					return errors.WithHint(
+						pgerror.New(
+							pgcode.FeatureNotSupported,
+							fmt.Sprintf("column %s of type %s is not allowed as the last column in an inverted index", col.Name, col.Type.Name()),
+						),
+						"see the documentation for more information about inverted indexes",
```

Comment:
"The last column of an inverted index definition must be one of the inverted indexable types: JSON, ARRAY, GEOMETRY, .... See the documentation on INVERTED INDEX for more details."